android/services/mongodb-remote/src/main/java/com/mongodb/stitch/android/services/mongodb/remote/RemoteMongoCollection.java

```diff
@@ -241,6 +246,22 @@
       final Bson update,
       final RemoteUpdateOptions updateOptions);
 
+
+  /**
+   * Watches specified IDs in a collection.  This convenience overload supports the use case
+   * of non-{@link BsonValue} instances of {@link ObjectId}.
+   * @param ids unique object identifiers of the IDs to watch.
+   * @return the stream of change events.
+   */
+  Task<Stream<ChangeEvent<DocumentT>>> watch(final ObjectId... ids);
+
+  /**
+   * Watches specified IDs in a collection.
+   * @param ids the ids to watch.
+   * @return the stream of change events.
+   */
+  Task<Stream<ChangeEvent<DocumentT>>> watch(final BsonValue... ids);
```

Comment:
[major] This pattern will not play nicely on Android.

A traditional listener pattern here would be more idiomatic. Core `Stream`'s `nextEvent` is also blocking, which is a no-go on Android.

What we're doing on iOS (and similarly JS) is to return a "session" that can invalidate the stream if need be, and that session acts as a shim between the delegate/listener and the stream itself.

Something like:
```java
WatchSession watch(final WatchChangeEventListener listener, final BsonValue... ids);

interface WatchChangeEventListener extends ChangeEventListener {
    void onStreamOpen();
    void onStreamClosed();
}

interface WatchSession {
    void invalidateStreamAndClose();
}

class WatchSessionImpl<DocumentT> implements WatchSession {
   WatchSessionImpl(Stream<DocumentT> stream, WatchChangeEventListener listener) {
       ....
   }
} 
```

The current `Stream` type would also need to be more reactive like Swift–  this is a flaw in the original design. The stream should have callbacks for `onOpen`, `onClosed`, `onEvent`. These should be bubbled up to the session, which then dispatches it to the listeners.

This is going to be required for the Sync cadence ticket as well. Streaming should be entirely asynchronous.

Reply:
done -- callbacks aren't as idiomatic for Java as they are for Swift, so I've reworked the stream to return new `Task`s for each event instead of synchronously returning events for now -- a callback mechanism can be built on top of this at the app level for now, but I have a feeling this all may need to be reworked anyway for reactive syncing